lib/DocumentManagerInterface.php

```diff
@@ -122,4 +124,56 @@ public function createQuery($query, $locale = null, array $options = []);
      * http://doctrine-phpcr-odm.readthedocs.org/en/latest/reference/query-builder.html
      */
     public function createQueryBuilder();
+
+    /**
+     * Retrieve the document inspector.
+     *
+     * NOTE: In the future this will be a factory for retrieving inspectors
+     * for a given object/class.
+     *
+     * @return DocumentInspector
+     */
+    public function getInspector();
+
+    /**
+     * Return the node manager instance.
+     *
+     * @return NodeManager
+     */
+    public function getNodeManager();
+
+    /**
+     * Return the document registry instance.
+     *
+     * @return DocumentRegistry
+     */
+    public function getRegistry();
+
+    /**
+     * Return the metadata factory instance.
+     *
+     * @return MetadataFactoryInterface
+     */
+    public function getMetadataFactory();
+
+    /**
+     * Return the event dispatcher instance.
+     *
+     * @return EventDispatcher
+     */
+    public function getEventDispatcher();
+
+    /**
+     * Return the PHPCR session instance.
+     *
+     * @return SessionInterface
+     */
+    public function getSession();
+
+    /**
+     * Return the proxy factory instance.
+     *
+     * @return ProxyFactory
+     */
+    public function getProxyFactory();
```

Comment:
Is it really necessary to put all of these into this interface? Until now I thought as the `DocumentManager` as a facade for a more complex underlying system. This way we populate quite a lot of these complex classes, making later refactorings pretty hard.